src/app/edit-maprule/preset/preset.component.ts

```diff
@@ -0,0 +1,173 @@
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, ViewEncapsulation, AfterViewInit, ViewChildren, QueryList, SimpleChange, SimpleChanges } from '@angular/core';
+import { Validators, FormGroup, FormControl, FormBuilder, FormArray, Form } from '@angular/forms';
+
+import { Field } from '../../shared/interfaces/field.interface';
+// import { FieldConfig } from '../../shared/interfaces/field-config.interface';
+import { FieldConfigService } from '../../core/services/field-config.service';
+import { NgbPanelChangeEvent } from '@ng-bootstrap/ng-bootstrap';
+import { Event } from '@angular/router';
+import { AccordionService } from '../../core/services/accordion.service';
+import { MapRulesService } from 'src/app/core/services/maprules.service';
+import { EditMapRuleComponent } from '../edit-maprule.component';
+
+@Component({
+  selector: 'app-preset',
+  styleUrls: [
+    './preset.component.css',
+    '../../shared/components/content.group.css',
+    '../form-table.css'
+  ],
+  templateUrl: './preset.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class PresetComponent {
+
+  @ViewChildren("preset") presets: QueryList<any>
+
+  // geometries: String[]
+  constructor(
+    private fb: FormBuilder,
+    private editMapRule: EditMapRuleComponent,
+    private fieldConfig: FieldConfigService,
+  ) { }
+
+  /* GETTERS FOR NESTED PARTS OF PRESET GROUP */
+  primaryFormArray(presetIndex: number): FormArray {
+    return this.editMapRule.presets.at(presetIndex).get('primary') as FormArray;
+  }
+
+  geometryArray(presetIndex: number): FormArray {
+    return this.editMapRule.presets.at(presetIndex).get('geometry') as FormArray;
+  }
+
+  fieldsFormArray(presetIndex: number): FormArray {
+    return this.editMapRule.presets.at(presetIndex).get('fields') as FormArray;
+  }
+
+  geometries(presetIndex: number): Array<any> {
+    let geometries = this.editMapRule.presetGeometries[presetIndex];
+    return this.fieldConfig.geom.map(function (g) {
+      return {
+        name: g,
+        checked: geometries.includes(g)
+      }
+    })
+  }
+
+  updatePresetGeometries(event: any, presetIndex: number) {
+    // if ()
+    let presetGeoms = this.editMapRule.presetGeometries[presetIndex];
+    let geometry = event.currentTarget.labels[0].innerText.trim();
+    if (event.currentTarget.checked) {
+      this.editMapRule.presetGeometries[presetIndex] = presetGeoms
+        .concat(geometry).sort()
+    } else {
+      this.editMapRule.presetGeometries[presetIndex] = presetGeoms
+        .filter(function (g) { return g !== geometry; })
+        .sort();
+    }
+  }
+
+  ngOnInit() {}
+
+  ngAfterViewInit() {
+    this.presets.changes.subscribe((presets: QueryList<any>) => {
+      if (!this.editMapRule.presets.value[presets.length - 1].presetName.length) {
```

Comment:
when we 'add new presets', this observable gets the current preset div refs, then figures out their open/closed state and what toggle buttons should look like.